core/templates/components/global-footer/global-footer.twig

```diff
@@ -1,29 +1,35 @@
-<footer{{ attributes }} class="global-footer {{ modifier_class }}">
-  <div class="global-footer__container">
-    <div class="global-footer__brand">
-      <a href="https://www.stanford.edu/" class="global-footer__logo">Stanford<br>University</a>
-    </div>
-    <div class="global-footer__content">
-      <nav role="navigation">
-        <ul class="global-footer__menu global-footer__menu--global">
-          <li><a href="https://www.stanford.edu">Stanford Home</a></li>
-          <li><a href="https://visit.stanford.edu/plan/">Maps &amp; Directions</a></li>
-          <li><a href="https://www.stanford.edu/search/">Search Stanford</a></li>
-          <li><a href="https://emergency.stanford.edu">Emergency Info</a></li>
-        </ul>
-        <ul class="global-footer__menu global-footer__menu--policy">
-          <li><a href="https://www.stanford.edu/site/terms/" title="Terms of use for sites">Terms of Use</a></li>
-          <li><a href="https://www.stanford.edu/site/privacy/" title="Privacy and cookie policy">Privacy</a></li>
-          <li><a href="https://uit.stanford.edu/security/copyright-infringement" title="Report alleged copyright infringement">Copyright</a></li>
-          <li><a href="https://adminguide.stanford.edu/chapter-1/subchapter-5/policy-1-5-4" title="Ownership and use of Stanford trademarks and images">Trademarks</a></li>
-          <li><a href="http://exploredegrees.stanford.edu/nonacademicregulations/nondiscrimination/" title="Non-discrimination policy">Non-Discrimination</a></li>
-          <li><a href="https://www.stanford.edu/site/accessibility" title="Report web accessibility issues">Accessibility</a></li>
-        </ul>
-      </nav>
-      <div class="global-footer__copyright">
-        <span>&copy; Stanford University.</span>
-        <span>&nbsp; Stanford, California 94305.</span>
+<footer{{ attributes }} class="page-footer  {{ modifier_class }}">
```

Comment:
Unfortunately, this breaks our BEM naming conventions. The outer most wrapper must be the block for every component and the nested children elements follow the block__element naming convention. This change does bring up the slight inconsistency in our naming pattern with the original block level class name in that it should have been `su-global-footer` to have started with.

Reply:
So, would grandchild selectors be our friend here? Something like:

```
<footer{{ attributes }} class="su-page-footer  {{ modifier_class }}">
  <section class="su-page-footer__global-footer">
    <div class="su-page-footer__global-footer__container">
      ...
```

Would this work? Do we only want underscores to appear once in a selector name? I, of course, we should avoid too often ending up with a component structure that leads to something like `parent__child__grandchild__greatgrandchild`, but perhaps this is an example where the grandchild depth is OK.

BEM naming isn’t strictly tied to the DOM and doesn’t limit the nested levels of a descendent element, but is rather available to help us identify the relationship with the top level of a component,  in our case, the page footer. That said, the above might work (?).

Whatcha think? Am I crazy?

Reply:
To be honest, I think we instead of refactoring the component, we simply remove the stickiness from its default state and make it a variant instead. This would keep the component consistent to where it is now and allow the implementor to add a css class to whatever element they want 'stuck' to the bottom. See https://github.com/SU-SWS/decanter/pull/300 for example.